runtime/lua/hl_yank.lua

```diff
@@ -0,0 +1,29 @@
+--- Highlight the yanked region
+--
+--- use from init.vim via
+---   au TextYankPost * lua require'hl_yank'()
+--- customize highlight group and timeout via
+---   au TextYankPost * lua require'hl_yank'("IncSearch", 500)
+-- @param higroup highlight group for yanked region
+-- @param timeout time in ms before highlight is cleared
+-- @param event event structure
+return function(higroup, timeout, event)
```

Comment:
This limits potentially adding more API later. Shouldn't we stick with the
simple pattern of returning a table of functions, rather than making the module
callable?

IIRC `require("inspect")` works like you have done here, but we don't expose
that module offically, we expose it as `vim.inspect`.

Reply:
This is just for the hl_yank function, which seems kinda specialized and one-shot. If I should put it in a different module, we can do that, of course -- but which one?

Reply:
If we want to you, you can make this return a table, with a metamethod `__call` that does this (I think that's the right metamethod). So we can do it like this and then later expose more if we wanted to.

I had suggested this initially because I liked doing `require('hl_yank')()` or similar.

Reply:
> you can make this return a table

^ yes, that is what I am suggesting. `require('foo')()` seems clever for no reason, and limits future expansion as I mentioned.

Reply:
Not sure what this could be expanded with? If you think this could be the germ of some module providing common functionality, I'd appreciate a suggestion. (A "highlight-yank" module makes no sense to me.)

Reply:
I'm saying, if you make it return a table later, you can make the returned table have the metamethod `__call` and it will be backwards compat.

There will be no functional difference for users if we decide to do that later (they will not have to change their configs).

Reply:
The difference is that we then have multiple conventions for modules. Why?

> Not sure what this could be expanded with?

That is never relevant. Trying to predict the future leads to bad APIs.

Reply:
But it's currently not a module -- it's a one-off function. I'm happy to turn it into a module, but _not_ into a `highlight-yank` module -- that's too finegrained. (Of course, in the end it's your call, if you insist.)

Reply:
Agreed on trying to predict future = bad.

I'm fine w/ returning a table, and exposing the function as `highlight_region` or something like that. Not to particular about the name.

Reply:
Doesn't matter much ATM. But I would like to enable this by default, and modeling it as a "plugin" is likely the most conventional way to do that.

Reply:
OK, then I would suggest making this into (the start of) a general `highlight.lua` module, with `hightlight_yank` as the (for now only) exported function.